### ISSUE TITLE
Missing parent property for identifiers of scope references

### ISSUE THREAD

**Issue by DMartens** [typescript-eslint/typescript-eslint#4705]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**
```TS
interface Foo {}
interface Bar implements Foo {}
```
[Playground Link](https://typescript-eslint.io/play/#ts=4.6.2&sourceType=module&showAST=es&code=JYOwLgpgTgZghgYwgAgGIHt3IN4F8BQoksiKAQnFMsALYAOANhDROAM5qY4FA&rules=N4XyA&tsConfig=N4XyA)

**Expected Result**
Expected the identifier of the reference of `Foo` in `interface Bar implements Foo` to have a `parent` property.

**Actual Result**
The identifier of the reference only has the properties `type`, `name`, `range` and `loc`.

**Additional Info**
I came across this bug when smoke testing an eslint rule.
When the `implements` is changed to `extends`, the reference identifier has a `parent` property.

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/scope-manager`      | `5.15.0` |
| `TypeScript`                            | `4.6.2` |
| `node`                                  | `17.7.2` |

**Comment by bradzacher** [comment]
This is working as intended.
ESLint is the one that adds parent pointers, not our tooling.

So if you're using the scope analyser outside of ESLint you will need to set your own parent pointers.

**Comment by DMartens** [comment]
I kept the reproduction code as short as possible. I am seeing this behaviour in the eslint context.
Full reproduction code:
```javascript
const { RuleTester } = require('eslint');

const ruleTester = new RuleTester();
const rule = {
	create(context) {
		return {
			'Program'() {
				const [reference] = context.getScope().references;
				console.log(Object.hasOwn(reference.identifier, 'parent'));
			},
		};
	},
};

ruleTester.run('rule', rule, {
	valid: [{
		parser: require.resolve('@typescript-eslint/parser'),
		code:   [
			'interface Foo {}',
			'interface Bar implements Foo {}', // No parent
		].join('\n'),
	}, {
		parser: require.resolve('@typescript-eslint/parser'),
		code:   [
			'interface Foo {}',
			'interface Bar extends Foo {}', // Has parent
		].join('\n'),
	}],
	invalid: [],
});
```
